// datastore-core/cjs/src/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var base = require('./base.js');
var memory = require('./memory.js');
var keytransform = require('./keytransform.js');
var sharding = require('./sharding.js');
var mount = require('./mount.js');
var tiered = require('./tiered.js');
var namespace = require('./namespace.js');
var errors = require('./errors.js');
var shard = require('./shard.js');
var shardReadme = require('./shard-readme.js');



exports.BaseDatastore = base.BaseDatastore;
exports.MemoryDatastore = memory.MemoryDatastore;
exports.KeyTransformDatastore = keytransform.KeyTransformDatastore;
exports.ShardingDatastore = sharding.ShardingDatastore;
exports.MountDatastore = mount.MountDatastore;
exports.TieredDatastore = tiered.TieredDatastore;
exports.NamespaceDatastore = namespace.NamespaceDatastore;
exports.abortedError = errors.abortedError;
exports.dbDeleteFailedError = errors.dbDeleteFailedError;
exports.dbOpenFailedError = errors.dbOpenFailedError;
exports.dbWriteFailedError = errors.dbWriteFailedError;
exports.notFoundError = errors.notFoundError;
exports.NextToLast = shard.NextToLast;
exports.PREFIX = shard.PREFIX;
exports.Prefix = shard.Prefix;
exports.README_FN = shard.README_FN;
exports.SHARDING_FN = shard.SHARDING_FN;
exports.ShardBase = shard.ShardBase;
exports.Suffix = shard.Suffix;
exports.parseShardFun = shard.parseShardFun;
exports.readShardFun = shard.readShardFun;
exports.readme = shardReadme;
